### ISSUE TITLE
UMLet Eclipse Plugin crashes on right click or drop down

### ISSUE THREAD

**Issue by Equiphract** [umlet/umlet#598]
I just installed the newest version (Date Updated: Wed, 2020-09-09 14:31)  of the UMLet Eclipse plugin. When I click on the palette drop down or right click anywhere Eclipse crashes and generates the following window:
![Screenshot from 2020-11-13 18-19-57](https://user-images.githubusercontent.com/31915427/99102711-c00d5a00-25de-11eb-995a-cec1350f9d5f.png)

I am using Linux Mint 20
This is the output of `uname -a`:
`Linux Hermes 5.4.0-53-generic #59-Ubuntu SMP Wed Oct 21 09:38:44 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux`

**Comment by 911992** [comment]
Am not really sure if it's related to Umlet, looks more like jvm level error. Have you tried with another jvm like v8 maybe?

**Comment by Equiphract** [comment]
I just tried switching to v8 but then eclipse wouldn't even start. Apparently it requires at least v11 to run.
This would be weird anyways as I have run the standalone UMLet version with this exact JRE version before...

I am using Eclipse 2020-09 by the way.

**Comment by 911992** [comment]
Yes, my bad(sorry) Eclipse 2020-09 needs j11 or above.

I just tried to run the same eclipse version on my Eclipse (same 2020-09) version, and it's working without no problem. I just tried your custom chen diagrams too, no issues! But my machine is windows, with jdk 14.

Please share the eclipse [crash](https://wiki.eclipse.org/FAQ_Where_can_I_find_that_elusive_.log_file%3F) too if applicable, thanks.

Would you please try with jdk 14, or even perform a clean install on eclipse and plugin and try again. I'm still not really sure if it's related to umlet-plugin explicitly

**Comment by afdia** [comment]
Hi

I have just tried the following:
1. download latest eclipse ee 2020-09 from https://www.eclipse.org/downloads/packages/
2. install umlet from marketplace and restart eclipse
3. everything seems to work using AdoptOpenJDK (build 11.0.9+11) on Windows 10 (I have moved some elements from the palette to the diagramm, switched palette, ...)

Maybe try it out using a fresh EclipseEE from the above link to make sure the problem is not caused by some setting or other installed plugin.

Due to the multitude of Eclipse configurations out there some people have problems with the plugin and most of the times we cannot really reproduce the problem, because its typically cause by some combination of Eclipse+Java+OS+other plugins etc.
Maybe check our wiki under https://github.com/umlet/umlet/wiki/Common-Problems or as @911992 said, follow the link and try to see if there is a crash log.

One interesting part is that you can install the plugin but it crashes when you interact with the palette. I don't really remember anyone else having this issue (most people with problems don't get it to work at all). I guess some issues could be caused due to the way Umlet embedds a Swing GUI into the Eclipse SWT GUI, but it should not crash.

**Comment by Equiphract** [comment]
Here is the error log: https://pastebin.com/tDLfuP80

I will make a clean install tomorrow and then update this comment.

**Comment by 911992** [comment]
Exceptions (except the first session/one) are about a class `org.eclipse.jface.text.BadPartitioningException` where not available in classpath, a little weird.

Do you have any other shell/widget to try? maybe kde for example?

**Comment by MeTheOtherGuy** [comment]
Im having the same error as @Equipharact

Here is the error log: https://pastebin.com/DDzSP1wX

**Comment by 911992** [comment]
I'm thinking the related error kind of related to underlying OS widget, and eclipse SWT, not sure.

And also maybe the real cause of error is not logged, as it could be jvm level.

Could you @Equiphract and @MeTheOtherGuy please do following
* Using jvm flag `-XX:ErrorFile=./eclipse_umlet_err` when running eclipse (or any other file path)
    * Where eclipse is located , add above command at the end of `eclipse.ini` file
    * If the jvm level err is produced, please share the log file
* Run Eclipse with jdk 14 (try both Oracle hotspot/openjdk)
* Use another OS/widget toolkit like gtk(over gnome)

Thanks

**Comment by Equiphract** [comment]
Ok here's what I've tried so far:
- I've put the jvm flag at the bottom of eclipse.ini and then let eclipse crash clicking the panel drop down
  - No error log generated
  - I tried multiple paths: the absolute path to my users home directory and the path @911992 suggested (which should generate the error log in the eclipse directory if I understood this correctly).
- I've installed the openjdk-14, made sure my system set it as default using the `update-alternatives` command and tried it again. Same crash, no error log.
- I've installed Xfce and tried it yet again but it still crashes and doesn't generate an error log.

The only thing I haven't yet tried is installing this Oracle hotspot thing because I don't yet know what it is and I haven't had the time to figure it out. But to be honest I doubt it would make a difference.

**Comment by 911992** [comment]
Thanks for tries @Equiphract 

> I've installed the openjdk-14...

You may use `-vm` eclipse-level(before `-vmargs`) arg inside the `eclipse.ini` to force it load the jvm lib from a specific path, rather than `PATH`, to make sure it loads the correct jvm, [doc here](https://wiki.eclipse.org/Eclipse.ini).

> this Oracle hotspot ...

Oracle hotspot is Oracle's impl of jvm(kind of privately in house), and known as kind of most stable jvm out there, it usually has less issues vs. openjdk. It's free(but has some craze licensing stuffs), and could be grabbed from [here](https://www.oracle.com/java/technologies/javase-downloads.html).

I would ask you please share any new logs you get, so we might be some luck to find the issue. Thanks.

**Comment by 911992** [comment]
Please also perform these steps, hope we find some luck/hope

As explained [here](https://stackoverflow.com/questions/11310970/eclipse-juno-doesnt-start), and [here](https://www.ixiasoft.com/documentation/IXIASOFT_CCMS/5.0/User_Guides_Advanced_User_DRM/chi1444319242316_2.html)

Thanks

**Comment by Equiphract** [comment]
> Please also perform these steps, hope we find some luck/hope
> 
> As explained [here](https://stackoverflow.com/questions/11310970/eclipse-juno-doesnt-start), and [here](https://www.ixiasoft.com/documentation/IXIASOFT_CCMS/5.0/User_Guides_Advanced_User_DRM/chi1444319242316_2.html)
> 
> Thanks

Did not do anything for me sadly...
Will try Oracle's JDK later today.

**Comment by Equiphract** [comment]
I just realised by the way: When I start eclipse from the terminal and open an UMLet plugin editor (the editor where one can create UMLet diagrams) the following lines are generated:
```
log4j:WARN No appenders could be found for logger (com.baselet.plugin.MainPlugin).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
```

And when I let eclipse crash this line gets generated:
`Inconsistency detected by ld.so: dl-lookup.c: 111: check_match: Assertion 'version->filename == NULL || ! _dl_name_match_p (version->filename, map)' failed!`

Does this give us some clues on what might be happening?

**Comment by 911992** [comment]
> Does this give us some clues on what might be happening?

Hmm, as expected, native-level stuff.

`ld.so` is a native dyn lib, the problem is out of umlet plugin. This could be bug between SWT and underlying widget, or jvm itself.

Please:
* perform a complete clean/uninstall on jdk/jvm you have
* update your system (not sure if it's essential)

Try Oracle's Hotspot as mentioned, and hope this time you won't have the crash. relative issues [here](https://unix.stackexchange.com/questions/532054/program-crashes-with-the-error-inconsistency-detected-by-ld-so-dl-lookup-c-11) and [here](https://github.com/hharrison/java3d-core/issues/28)

**Comment by Equiphract** [comment]
When I use Oracle's jdk 15 and try to open an UMLet document in eclipse I only get a tiny tab with only the close-tab button appearing on it as you can see to the left of the `Config.java` tab in the screenshot below. The UMLet editor won't even appear when I click on that tab. Did I use the wrong jdk? In an earlier reply you told me to use the jdk 14 but I wasn't able to find such a version on [Oracle's download page](https://www.oracle.com/java/technologies/javase-downloads.html).
![eclipse_umlet_bug](https://user-images.githubusercontent.com/31915427/100384454-7e3dd400-3020-11eb-999e-f2de0af56470.png)

Output of `java -version`:
![java_15_version](https://user-images.githubusercontent.com/31915427/100384605-ceb53180-3020-11eb-82c7-64ed70446b29.png)

**Comment by 911992** [comment]
Yes, it seems you are running hotspot, and I think having v15 is ok. even 11.

It seems that the problem *could* be related to eclipse or the plugin, we will see.

Would you mind please remove the eclipse plugin, and install it again.

Also if the same error/issue present, please share the eclipse log file as you did early, hope this time we find something relative.

**Comment by Equiphract** [comment]
Sorry for the delay, the issue still persists after reinstalling the plugin and while using Oracle's JVM.
Here is the error log: https://pastebin.com/F7XH36U6

**Comment by 911992** [comment]
Good, thanks for the log file, now things are very clear, and it's beyond umlet plugin, and sits at eclipse level.

[This](https://stackoverflow.com/a/15844683) is a similar issue you are having that could help.  
I'm not a linux expert, so cannot tell you where exactly you need to look, to change the default jvm lib path.

As suggested by the answer(link above), try
```bash
echo "-Djava.library.path=/usr/lib/jni/" >> eclipse.ini
```
where `eclipse.ini` is the same file was mentioned before.

But again, I'm not sure if that `/usr/lib/jni/` would work for your distro or not
